cmake/add_library_macros.cmake

```diff
@@ -167,9 +172,9 @@ function(openms_add_library)
   #------------------------------------------------------------------------------
   # Link library against other libraries
   if(openms_add_library_LINK_LIBRARIES)
-    ## check for consistent lib arch (e.g. all 64bit)?
+    ## check for consistent lib arch (e.g. all 64bit?)
     check_lib_architecture(openms_add_library_LINK_LIBRARIES)
-    target_link_libraries(${openms_add_library_TARGET_NAME} ${openms_add_library_LINK_LIBRARIES})
+    target_link_libraries(${openms_add_library_TARGET_NAME} PRIVATE ${openms_add_library_LINK_LIBRARIES})
```

Comment:
It could be, yes. I will investigate. In the end I want to make as much PRIVATE as possible, such that libraries that are only used and not interfaces by us are not passed to libraries linking against us.